api/tests/validate_legacy_soap_api/validate_soap_endpoints.py

```diff
@@ -0,0 +1,183 @@
+    if simpler_resp is None:
+        raise AssertionError("No simpler response found")
+    simpler_xml_response = simpler_resp.to_flask_response()[0]
+    simpler_match = re.search(
+        r"<soap:Envelope.*</soap:Envelope>", simpler_xml_response.decode(), re.DOTALL
+    )
+    if not simpler_match:
+        raise AssertionError("No xml found")
+
+    """
+    Compare the cleaned xml output from the legacy endpoint to the cleaned xml generated by simpler
+    """
+    pattern = r"tracking number:(.*?)\)"
+    simpler_xml = simpler_match.group(0)
+    legacy_xml = legacy_match.group(0)
+    cleaned_legacy = re.sub(pattern, "tracking number:", legacy_xml, flags=re.DOTALL)
+    cleaned_simpler = re.sub(pattern, "tracking number:", simpler_xml, flags=re.DOTALL)
+    assert cleaned_legacy == cleaned_simpler
+
+
+def validate_grantors_get_application_zip_request(db_session: db.Session) -> None:
+    simpler_tracking_number = "GRANT80000001"
+    soap_request = SOAPRequest(
+        api_name=SimplerSoapAPI.GRANTORS,
+        method="POST",
+        full_path=SOAP_URI,
+        headers=HEADERS,
+        data=get_grantors_get_application_zip_request_data(simpler_tracking_number),
+        operation_name="GetApplicationZipRequest",
+    )
+    simpler_resp = get_simpler_soap_response(
+        soap_request, SOAPResponse(data=b"", headers=HEADERS, status_code="500"), db_session
+    )
+    assert simpler_resp.status_code == 200
+    simpler_xml = simpler_resp.to_flask_response()[0]
+    # Currently will return nothing but this should be how a test validates this response xml
+    assert simpler_xml != b""
+    simpler_match = re.search(r"<soap:Envelope.*</soap:Envelope>", simpler_xml.decode(), re.DOTALL)
+    assert simpler_match is not None
+    simpler_tree = etree.fromstring(simpler_match.group(0))
+    operation_element = next(
+        simpler_tree.iterfind(
+            ".//{http://apply.grants.gov/services/AgencyWebServices-V2.0}GetApplicationZipResponse"
+        )
+    )
+
+    # XOP element removed due to it being a pointer to where data is stored not part of a model defined in the xsd
+    NS_XOP = {"xop": "http://www.w3.org/2004/08/xop/include"}
+    file_handler_elements = operation_element.findall(
+        ".//ns:FileDataHandler",
+        namespaces={"ns": "http://apply.grants.gov/services/AgencyWebServices-V2.0"},
+    )
+    for handler in file_handler_elements:
+        xop_include = handler.find("xop:Include", namespaces=NS_XOP)
+        handler.remove(xop_include)
+        handler.text = None
+
+    schema_tree = etree.parse("tests/legacy_soap_api/xsd/AgencyWebServices-V2.0.xsd")
+    schema_validator = etree.XMLSchema(schema_tree)
+    schema_validator.assertValid(operation_element)
+    zip_buffer = io.BytesIO(simpler_xml)
+    with zipfile.ZipFile(zip_buffer, "r") as zip_file:
+        assert zip_file.testzip() is None
+
+
+@click.command()
+def validate_simpler_endpoints() -> None:
+    print("=start=")
```

Comment:
Could we use either logging or print statements? If you want to control the formatting to be more than just text, print statements are fine, although for errors you might want to make them color-coded which is a bit trickier.

Reply:
switching to logging.